src/main/java/tdl/s3/sync/FolderSynchronizer.java

```diff
@@ -3,12 +3,15 @@
 import tdl.s3.upload.FileUploadingService;
 
 import java.nio.file.Path;
+import java.util.List;
 
 public class FolderSynchronizer {
 
-    private FolderScanner folderScanner;
+    private final FolderScanner folderScanner;
 
-    private FileUploadingService fileUploadingService;
+    private final FileUploadingService fileUploadingService;
+
+    private List<SyncProgressListener> listeners;
```

Comment:
Use one single instance instead of a List. If the user needs multiple listeners it can implement the interface and wrap multiple listeners.